src/submission/submission-api.ts

```diff
@@ -88,9 +92,38 @@ class SubmissionController {
     await submission.operations.deleteRegistration(registrationId, programId);
     return res.status(200).send();
   }
+
+  @HasSubmittionAccess((req: Request) => req.params.programId)
+  async saveClinicalTsvFile(req: Request, res: Response) {
+    if (!isValidCreateBody(req, res, req.params.clinicalType)) {
+      return;
+    }
+    const file = req.file;
+    let records: ReadonlyArray<Readonly<{ [key: string]: string }>>;
+    try {
+      records = await TsvUtils.tsvToJson(file.path);
+    } catch (err) {
+      return ControllerUtils.badRequest(res, {
+        msg: `failed to parse the tsv file: ${err}`,
+        code: ErrorCodes.TSV_PARSING_FAILED
+      });
+    }
+    res.set("Content-Type", "application/json");
+    const command: SaveClinicalCommand = {
+      records: records,
+      programId: req.params.programId,
+      clinicalType: req.params.clinicalType
+    };
+    const result = await submission.operations.uploadClinical(command);
+    console.log(result);
```

Comment:
there is a Logger L.debug(xxx)

Reply:
I don't even remember why I logged this.